src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -182,20 +157,18 @@ protected override async Task OnInitializedAsync()
         async Task SubscribeResourcesAsync()
         {
             var preselectedVisibleResourceTypes = VisibleTypes?.Split(',').ToHashSet();
+            var preselectedVisibleResourceStates = VisibleStates?.Split(',').ToHashSet();
+            var preselectedVisibleResourceHealthStates = VisibleHealthStates?.Split(',').ToHashSet();
 
             var (snapshot, subscription) = await DashboardClient.SubscribeResourcesAsync(_watchTaskCancellationTokenSource.Token);
 
             // Apply snapshot.
             foreach (var resource in snapshot)
             {
                 var added = _resourceByName.TryAdd(resource.Name, resource);
-
-                _allResourceTypes.TryAdd(resource.ResourceType, true);
-
-                if (preselectedVisibleResourceTypes is null || preselectedVisibleResourceTypes.Contains(resource.ResourceType))
-                {
-                    _visibleResourceTypes.TryAdd(resource.ResourceType, true);
-                }
+                _resourceTypesToVisibility.TryAdd(resource.ResourceType, preselectedVisibleResourceTypes is null || preselectedVisibleResourceTypes.Contains(resource.ResourceType));
+                _resourceStatesToVisibility.TryAdd(GetStateOrDefaultText(resource.State, Loc), preselectedVisibleResourceStates is null || preselectedVisibleResourceStates.Contains(resource.State ?? string.Empty));
+                 _resourceHealthStatusesToVisibility.TryAdd(GetStateOrDefaultText(resource.HealthStatus?.Humanize(), Loc), preselectedVisibleResourceHealthStates is null || preselectedVisibleResourceHealthStates.Contains(resource.HealthStatus?.Humanize() ?? string.Empty));
```

Comment:
If there is no value then the key can be an empty string. It shouldn't be translated text.

Translated text should only be applied in the UI, and this isn't UI related code.